File: ScienceArchive.Server/ScienceArchive.Api/Responses/Response.cs
using System;
namespace ScienceArchive.Api.Responses
{
    public abstract class Response
    {
        public Response(object? data = null)
        {
            Data = data;
        }

        /// <summary>
        /// Indicates if request was successful
        /// </summary>
        public bool Success { get; set; }

        /// <summary>
        /// Data of the response
        /// </summary>
        public object? Data { get; set; }

        /// <summary>
        /// Error occured while request
        /// </summary>
        public string? Error { get; set; }
    }
}

File: science-archive-web/src/app/models/responses/response.ts
/**
 * Represent response from API
 */
export interface Response<T> {
  /**
   * Success status of response
   */
  success: boolean;

  /**
   * Data returned from API
   */
  data: T | null;

  /**
   * Error accured while processing request
   */
  error: string | null;
}
